### PR TITLE
Isolated cert generator

### PR DESCRIPTION
**Pull request by alexjh**
Add a cert generator container with our support scripts so customers can generate their own certs.

### CODE REVIEW

File: make/cert-generator [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+certgen_docker_root=${certgen_docker_root:-$(mktemp -d)}
+
+cp "${GIT_ROOT}/docker-images/cert-generator/Dockerfile" "${certgen_docker_root}"
+cp "${GIT_ROOT}/docker-images/cert-generator/cert-generator.sh" "${certgen_docker_root}"
+cp "${GIT_ROOT}/bin/generate-dev-certs.sh" "${certgen_docker_root}"
+cp "${GIT_ROOT}/src/uaa-fissile-release/generate-certs.sh" "${certgen_docker_root}"
+
+docker build -t splatform/cert-generator "${certgen_docker_root}"
+
+rm -rf "${certgen_docker_root}"
```
Comment (mook-as) [inline make/cert-generator]: This should done via a `trap`, so that it gets deleted on failure too

File: bin/generate-dev-certs.sh [hunk 2]
```diff
@@ -226,16 +223,13 @@ mv -f ${internal_certs_dir}/consul_agent.csr ${internal_certs_dir}/agent.csr
...
-#            (based on the length of the fingerprint without type info).
-#            And cf requires MD5, does not like SHA256.
+# ATTENTION: Generate the fingerprint in MD5 format
 #
 # See  https://github.com/cloudfoundry/diego-release/tree/develop/examples/aws#generating-ssh-proxy-host-key-and-fingerprint
 # and  https://github.com/cloudfoundry/cli/issues/817
 #
 ssh-keygen -b 4096 -t rsa -f "${certs_path}/app_ssh_key" -q -N "" -C hcf-ssh-key
-ssh-keygen -E MD5 -lf "${certs_path}/app_ssh_key" | awk '{print $2}' | sed -e 's/^MD5://' > "${certs_path}/app_ssh_host_key_fingerprint"
+awk '{print $2}' "${certs_path}/app_ssh_key.pub" | base64 -d | md5sum -b | sed 's/../&:/g; s/: .*$//' > "${certs_path}/app_ssh_host_key_fingerprint"
```
Comment (alexjh) [inline bin/generate-dev-certs.sh]: Fixed in 0ede6157

File: docker-images/cert-generator/cert-generator.sh [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+DOMAIN=${DOMAIN}
+HCP_SERVICE_DOMAIN_SUFFIX=${namespace}.svc.cluster.local
+
+EOF
+
+/generate-certs.sh -e /tmp/scf-env /tmp/uaa-certs.env > /dev/null
+/generate-dev-certs.sh -e /tmp/scf-env "${namespace}" /tmp/scf-certs.env > /dev/null
+
+sed 's/^\([A-Z_]\+\)=\(.\+\)/\1: "\2"/g' < /tmp/uaa-certs.env > /out/uaa-cert-values.yaml
```
Comment (alexjh) [inline docker-images/cert-generator/cert-generator.sh]: Fixed in 091fe5f8

File: make/cert-generator (hunk 1 above)
Comment (alexjh) [inline make/cert-generator]: Fixed in 404122e0

File: docker-images/cert-generator/cert-generator.sh [hunk 4]
```diff
@@ -0,0 +1,19 @@
+#!/bin/sh
+set -e
+
+export PATH=$PATH:/root/go/bin
+
+mkdir -p /tmp/scf-env
```
Comment (alexjh) [inline docker-images/cert-generator/cert-generator.sh]: Fixed in 4cd2292a

File: docker-images/cert-generator/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,13 @@
+FROM opensuse
+
+RUN zypper --non-interactive addrepo --gpgcheck --refresh --priority 120 --check obs://devel:languages:go devel:languages:go && \
+    zypper --non-interactive --gpg-auto-import-keys refresh && \
+    zypper --non-interactive install --no-confirm --from devel:languages:go go && \
+    zypper --non-interactive install --no-confirm git vim && \
+    go get github.com/square/certstrap && \
+    zypper clean -a
+
+RUN mkdir /scf-settings /uaa-settings /out
```
Comment (alexjh) [inline docker-images/cert-generator/Dockerfile]: Fixed in a3825467

/*-settings weren't needed any more.

File: docker-images/cert-generator/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,13 @@
+FROM opensuse
+
+RUN zypper --non-interactive addrepo --gpgcheck --refresh --priority 120 --check obs://devel:languages:go devel:languages:go && \
+    zypper --non-interactive --gpg-auto-import-keys refresh && \
+    zypper --non-interactive install --no-confirm --from devel:languages:go go && \
+    zypper --non-interactive install --no-confirm git vim && \
+    go get github.com/square/certstrap && \
```
Comment (alexjh) [inline docker-images/cert-generator/Dockerfile]: Might as well clean it up. Fixed in 404122e0

File: docker-images/cert-generator/cert-generator.sh [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+cat <<EOF > /tmp/scf-env/network.env
+
+DOMAIN=${DOMAIN}
+HCP_SERVICE_DOMAIN_SUFFIX=${namespace}.svc.cluster.local
```
Comment (alexjh) [inline docker-images/cert-generator/cert-generator.sh]: Fixed in 4866f17e

**Pull request merged**